woocommerce/woocommerce-gutenberg-products-block
# Add Notices Documentation to Storybook and upgrade to Storybook 7

## What

This PR upgrades Storybook to v7 and adds usage guidelines in the form of MDX files for NoticeBanner and SnackbarList components. It was necessary to bump versions of several packages to support Storybook 7, and modify existing stories for compatibility. 

As well as the upgrade and new MDX files, I reorganised existing stories so that the sidebar is easier to navigate. I've gone through each story to ensure there are no errors.

Fixes #10681

## Why

Storybook 7 has improved UI and supports inline documentation (MDX files) for documenting our components with markdown.

## Testing Instructions

1. Install npm dependencies and successfully build blocks. There should be no errors with the build.
2. Run `npm run storybook` and check for successful build.
3. Navigate to storybook. Go through each item in the navigation and check for errors.
4. Go to Base Components > NoticeBanner > Docs. proof.
5. Go to Base Components > SnackbarList > Docs. proof.


## Screenshots or screencast

![Screenshot 2023-10-31 at 15 44 04](https://github.com/woocommerce/woocommerce-blocks/assets/90977/d654f2ea-3457-4b48-9829-71bb819bd617)
![Screenshot 2023-10-31 at 15 43 53](https://github.com/woocommerce/woocommerce-blocks/assets/90977/80d94919-7205-4bfd-ac89-df133c47bbd7)

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog

N/A
